### PR TITLE
imx6ull-flash: add read/write/erase test with ecc

### PR DESCRIPTION
**Pull request by jsarzy**
## Description
Add write/read/erase with error checking

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: imx6ull.

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: storage/imx6ull-flash/imx6ull-flash-test.c [hunk 1]
```diff
@@ -1006,6 +1006,73 @@ void test_single_block_raw(flashdrv_dma_t *dma, uint32_t blockno, uint8_t *data,
...
+		}
+	}
+
+	for (i = 0; i < flashinfo.writesz; ++i) {
+		if (data[i] != byte)
+			printf("[%4u] write/read(0x%02x)[%u] invalid data: 0x%02x\n", blockno, byte, i, data[i]);
+	}
+}
+
+void test_single_block(flashdrv_dma_t *dma, uint32_t blockno, uint8_t *data, uint8_t *meta)
```
Comment (nalajcie) [inline storage/imx6ull-flash/imx6ull-flash-test.c:1055]: I know this is aligned with how tests are written currently but I would have preferred for test to return if it passed or failed.

To be considered in the future.

**Review by nalajcie — APPROVED** [review]
added some ideas which might be implemented in the future.

File: storage/imx6ull-flash/imx6ull-flash-test.c [hunk 2]
```diff
@@ -1006,6 +1006,73 @@ void test_single_block_raw(flashdrv_dma_t *dma, uint32_t blockno, uint8_t *data,
...
+{
+	int err;
+	uint32_t addr = blockno * 64;
+
+	if ((err = flashdrv_erase(dma, addr) < 0)) {
+		printf("[%4u] erase(1) failed: %d\n", blockno, err);
+		/* assume bad-block, don't test further */
```
[the remark this answers is not in the document]
Comment (jsarzy) [reply storage/imx6ull-flash/imx6ull-flash-test.c]: I add explicit bad block check

File: storage/imx6ull-flash/imx6ull-flash-test.c [hunk 3]
```diff
@@ -1006,6 +1006,73 @@ void test_single_block_raw(flashdrv_dma_t *dma, uint32_t blockno, uint8_t *data,
...
+		return;
+	}
+
+	_write_and_check(dma, blockno, data, meta, 0x55);
+
+	if ((err = flashdrv_erase(dma, addr) < 0))
```
[the remark this answers is not in the document]
Comment (jsarzy) [reply storage/imx6ull-flash/imx6ull-flash-test.c:1072]: ack

File: storage/imx6ull-flash/imx6ull-flash-test.c (hunk 1 above)
Comment (jsarzy) [reply storage/imx6ull-flash/imx6ull-flash-test.c:1055]: ack

**Review by nalajcie — APPROVED** [review]

**Pull request merged**